### PR TITLE
Add .myvalenstate

### LINKED ISSUE #108: [Feature request] myvalenstate
Which country or state would be more befitting to me for a beautiful valentine, if could go there?

I would like you to write a bot, that based on my tag name, would find a state with the most matching characters. And that state will be my (probably) next valentine vacation spot.  

## Implementation details

- [ ] Write the bot command !myvalenstate
- [ ] Return a state with the most matching character with tag name (plus some comment).

## Example
.myvalenstate
Your valenState is `Sweden`
Sweden is a Scandinavian nation with thousands of coastal islands and inland lakes, along with vast boreal forests and glaciated mountains.
image

### PR DESCRIPTION
**Pull request by darthdelay**
Closes #108

This pull request adds a beta version of the command myvalenstate.

This command sends an embed containing a users valenstate to the channel
it has been called in. The process used here is to first put the username
into a spaceless, lower case form and than compare it to the countries
listed under valenstates.json, put into the same form, using an implementation
of the Levenshtein algorithm.

This beta version already implements everything needed for the command to
work properly. The follwing things are missing, not meant for the future
full implementation or are subject to change if needed:

- Short comments (25 missing)
- name parameter in myvalenstate (not meant for staying/Debug)
- Object names in valenstates.json (subject to change) (1)
- Docstring for levenshtein (subject to change)
- Docstring for myvalenstate (subject to change)

(1) - The object names can be put into the right case to omit the operation
        operation on line 54 `lower_state = state.lower().replace(' ', '')`

### CODE REVIEW

File: bot/seasons/valentines/myvalenstate.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import discord
+from discord.ext import commands
+
+from bot.constants import Colours
+
+log = logging.getLogger(__name__)
+
+with open(Path('bot', 'resources', 'valentines', 'valenstates.json'), 'r') as file:
+    STATES = json.load(file)
+    # STATES.sort()
```
Comment (sco1) [inline bot/seasons/valentines/myvalenstate.py]: Don't need the comment

File: bot/seasons/valentines/myvalenstate.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                value=f"You have {len(matches)} more matches, these being {leftovers}."
+            )
+        embed.set_image(
+            url=STATES[valenstate]["flag"]
+        )
+        await ctx.channel.send(embed=embed)
+
+
+def setup(bot):
+    bot.add_cog(MyValenstate(bot))
```
Comment (sco1) [inline bot/seasons/valentines/myvalenstate.py]: Need a logging statement for the cog load

File: bot/seasons/valentines/myvalenstate.py (hunk 1 above)
Comment (CyrolX) [reply bot/seasons/valentines/myvalenstate.py]: Done

File: bot/seasons/valentines/myvalenstate.py (hunk 2 above)
Comment (CyrolX) [reply bot/seasons/valentines/myvalenstate.py]: Added as `log.debug("MyValenstate cog loaded")`

File: bot/seasons/valentines/myvalenstate.py (hunk 1 above)
Comment (darthdelay) [reply bot/seasons/valentines/myvalenstate.py]: I dont know who this CyrolX is but he did a great job.

**Pull request merged**